File: Web/re-peta/src/main/java/com/example/prenotazioni/DAO/Prenotazione.java
package com.example.prenotazioni.DAO;

public class Prenotazione {
    private Ripetizione ripetizione;
    private Docente docente;
    private Utente utente;
    private String stato;

    public Prenotazione(Ripetizione ripetizione, Docente docente, Utente utente, String stato) {
        this.ripetizione = ripetizione;
        this.docente = docente;
        this.utente = utente;
        this.stato = stato;
    }

    public static Prenotazione defaultPrenotazione() {
        return new Prenotazione(null, null, null, null);
    }

    public Ripetizione getRipetizione() {
        return ripetizione;
    }

    public Docente getDocente() {
        return docente;
    }

    public Utente getUtente() {
        return utente;
    }

    public String getStato() {
        return stato;
    }

    @Override
    public String toString() {
        return "{Ripetizione." + ripetizione.simpleRipetizione() +
                "\tdocente = " + docente.simpleDocente() +
                "\tutente = " + utente.getUsername() +
                "\tstato = " + stato + "}\n";
    }

}

File: App/re_peta/lib/model/prenotazione.dart
import 'dart:convert';

import 'user.dart';
import 'docente.dart';
import 'ripetizione.dart';

class Prenotazione {
  Ripetizione ripetizione;
  Docente docente;
  User utente;
  String stato;

  Prenotazione(this.ripetizione, this.docente, this.utente, this.stato);

  factory Prenotazione.fromJson(dynamic js) {
    return Prenotazione(
        Ripetizione.fromJson(js['ripetizione']),
        Docente.fromJson(js['docente']),
        User.fromJson(js['utente']),
        js['stato'] as String);
  }

  Map<String, dynamic> toJson() => {
        'ripetizione': jsonEncode(ripetizione),
        'docente': jsonEncode(docente),
        'user': jsonEncode(utente),
        'stato': stato
      };

  @override
  String toString() {
    return "Prenotazione: corso = ${ripetizione.corso.titolo}\tdocente = ${docente.nome} ${docente.cognome}\tutente = ${utente.username}\tstato = $stato}\n";
  }

  static String printPrenotazioni(List<Prenotazione> prens) {
    String result = 'Prenotazioni: {';
    for (var prenotazione in prens) {
      result += "${prenotazione.toString()}\n";
    }
    return "$result \n}";
  }
}
